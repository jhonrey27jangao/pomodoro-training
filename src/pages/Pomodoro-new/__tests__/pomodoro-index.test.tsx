import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import PomodoroApp from "../index";
import { Formik } from "formik";

describe("Test for Index Pomodoro App", () => {
  const mockStore = configureStore();
  let store: any;
  let initialState: any;

  beforeEach(() => {
    initialState = {
      pomodoroReducer: {
        pomodoroDuration: 25,
        shortBreak: 10,
        longBreak: 15,
        prevId: 1,
        Lists: [
          {
            id: 1,
            title: "Taena hehe",
            savedPomodoro: 25,
            savedLongBreak: 15,
            savedShortBreak: 10,
            done: false,
            favorites: false,
          },
          {
            id: 2,
            title: "Taena KO",
            savedPomodoro: 25,
            savedLongBreak: 15,
            savedShortBreak: 10,
            done: true,
            favorites: true,
          },
        ],
      },
    };
    store = mockStore(initialState);
  });

  afterEach(() => {
    cleanup();
  });

  it("renders the component", () => {
    const { container, queryAllByText } = render(
      <Formik initialValues={initialState} onSubmit={() => {}}>
        {() => (
          <Provider store={store}>
            <PomodoroApp />
          </Provider>
        )}
      </Formik>
    );
    expect(container).toMatchSnapshot();
    /* 
    test play 
    */
    fireEvent.click(queryAllByText("❚❚")[0]);
    fireEvent.click(queryAllByText("▷")[0]);
    fireEvent.click(queryAllByText("↻")[0]);
    /*
     Add Todo
    */
    fireEvent.change(document.getElementById("todoName") as HTMLElement, {
      target: { value: "AddedTodo" },
    });

    fireEvent.click(queryAllByText("Add Todo")[0]);
    fireEvent.click(queryAllByText("Play")[0]);
    fireEvent.click(queryAllByText("Delete")[0]);
    /*
    trigger general buttons
    */
    fireEvent.click(queryAllByText("Auto")[0]);
    fireEvent.click(queryAllByText("Manual")[0]);
    fireEvent.click(queryAllByText("Pomodoro")[0]);
    fireEvent.click(queryAllByText("L-Break")[0]);
    fireEvent.click(queryAllByText("S-break")[0]);
    fireEvent.click(queryAllByText("About")[0]);
    fireEvent.click(queryAllByText("X")[0]);
    /*
    modify time settings
    */
    fireEvent.click(queryAllByText("Settings")[0]);
    fireEvent.click(queryAllByText("X")[0]);
    fireEvent.click(queryAllByText("Settings")[0]);
    fireEvent.change(document.getElementById("PDuration") as HTMLElement, {
      target: { value: "30" },
    });
    fireEvent.change(document.getElementById("LBreak") as HTMLElement, {
      target: { value: "20" },
    });
    fireEvent.change(document.getElementById("SBreak") as HTMLElement, {
      target: { value: "15" },
    });
    fireEvent.click(queryAllByText("Update")[0]);
    /*
     Drag Item
    */
    fireEvent.drag(queryAllByText("Taena hehe")[0]);
    fireEvent.dragOver(queryAllByText("Taena hehe")[0]);
    fireEvent.drop(queryAllByText("Taena hehe")[0]);
    /*
     Edit Todo
    */
    fireEvent.click(queryAllByText("✎")[0]);
    fireEvent.click(queryAllByText("X")[0]);
    fireEvent.click(queryAllByText("✎")[0]);
    fireEvent.click(queryAllByText("Done")[0]);
    fireEvent.click(queryAllByText("Added to favorites")[0]);
    fireEvent.change(document.getElementById("editTodoInput") as HTMLElement, {
      target: { value: "adsadsad" },
    });
    fireEvent.click(queryAllByText("Update Todo")[0]);
    /*
    Pick fav
    */
    fireEvent.click(queryAllByText("Show Favorites")[0]);
    fireEvent.click(queryAllByText("X")[0]);
    fireEvent.click(queryAllByText("Show Favorites")[0]);
    fireEvent.click(queryAllByText("Taena hehe")[0]);
  });
});
