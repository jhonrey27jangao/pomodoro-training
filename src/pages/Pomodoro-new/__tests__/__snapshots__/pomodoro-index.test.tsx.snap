// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Index Pomodoro App renders the component 1`] = `
<div>
  <form
    action="#"
  >
    <h2
      class="sc-AxhUy kUxjaF"
      type="h2"
    >
      <span
        style="font-family: 'Orbitron', sans-serif;"
      >
        Timer: 
        <br />
        25m:0s
      </span>
    </h2>
    <div
      class="sc-AxirZ iESHQS"
    >
      <button
        class="sc-fznKkj kWHrLL"
        disabled=""
        type="submit"
        width="100"
      >
        <span
          class="sc-fzplWN gcyZgM"
        >
          ❚❚
        </span>
      </button>
      <button
        class="sc-fznKkj kWHrLL"
        type="submit"
        width="100"
      >
        <span
          class="sc-fzplWN gcyZgM"
        >
          ▷
        </span>
      </button>
      <button
        class="sc-fznKkj kWHrLL"
        type="submit"
        width="100"
      >
        <span
          class="sc-fzplWN gcyZgM"
        >
          ↻
        </span>
      </button>
    </div>
    <div
      class="sc-AxjAm dNRMDR"
    >
      <div
        class="sc-AxirZ khsKjH"
      >
        <input
          class="sc-fzpans dJYYuJ"
          id="todoName"
          placeholder="Add todo"
          value=""
        />
        <button
          class="sc-fznKkj bHKJNM"
          type="submit"
        >
          <span
            class="sc-fzplWN gcyZgM"
          >
            Add Todo
          </span>
        </button>
      </div>
      <div
        class="sc-AxirZ dfLLgI"
      >
        <div
          style="padding: 47px 15px 0px;"
        >
          <button
            class="sc-fznKkj eboykN"
            type="submit"
            width="100"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              Auto
            </span>
          </button>
          <button
            class="sc-fznKkj kCeHpr"
            type="submit"
            width="120"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              Pomodoro
            </span>
          </button>
          <button
            class="sc-fznKkj kCeHpr"
            type="submit"
            width="120"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              L-Break
            </span>
          </button>
          <button
            class="sc-fznKkj kCeHpr"
            type="submit"
            width="120"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              S-break
            </span>
          </button>
          <button
            class="sc-fznKkj eboykN"
            type="submit"
            width="100"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              Settings
            </span>
          </button>
          <button
            class="sc-fznKkj eboykN"
            type="submit"
            width="100"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              About
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-AxiKw hWCQXE"
    >
      <button
        class="sc-fznyAO jbexnx"
        type="submit"
        width="150"
      >
        Show Favorites
      </button>
      
      <table>
        <thead>
          <tr>
            <th />
            <th>
              Name
            </th>
            <th>
              Duration
            </th>
            <th>
              L-break
            </th>
            <th>
              S-break
            </th>
            <th>
              Status
            </th>
            <th>
              Favorites
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class=""
            data-position="0"
            draggable="true"
          >
            <td>
              <p
                class="sc-Axmtr cuKjKQ"
                type="p"
              >
                ✎
              </p>
            </td>
            <td>
               
              Taena hehe
            </td>
            <td>
               
              25
            </td>
            <td>
               
              15
            </td>
            <td>
               
              10
            </td>
            <td>
              To do
            </td>
            <td>
               
              No
            </td>
            <td>
              <button
                class="sc-fznyAO jqcObA"
                type="submit"
                width="100"
              >
                Play
              </button>
              <button
                class="sc-fznyAO bLTcgU"
                type="submit"
                width="100"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr
            class="done"
            data-position="1"
            draggable="true"
          >
            <td>
              <p
                class="sc-Axmtr cuKjKQ"
                type="p"
              >
                ✎
              </p>
            </td>
            <td>
               
              Taena KO
            </td>
            <td>
               
              25
            </td>
            <td>
               
              15
            </td>
            <td>
               
              10
            </td>
            <td>
              Done
            </td>
            <td>
               
              Yes
            </td>
            <td>
              <button
                class="sc-fznyAO jqcObA"
                disabled=""
                type="submit"
                width="100"
              >
                Play
              </button>
              <button
                class="sc-fznyAO bLTcgU"
                type="submit"
                width="100"
              >
                Delete
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </form>
</div>
`;

exports[`Test for Index Pomodoro App renders the component without values on table 1`] = `
<div>
  <form
    action="#"
  >
    <h2
      class="sc-AxhUy kUxjaF"
      type="h2"
    >
      <span
        style="font-family: 'Orbitron', sans-serif;"
      >
        Timer: 
        <br />
        25m:0s
      </span>
    </h2>
    <div
      class="sc-AxirZ iESHQS"
    >
      <button
        class="sc-fznKkj kWHrLL"
        disabled=""
        type="submit"
        width="100"
      >
        <span
          class="sc-fzplWN gcyZgM"
        >
          ❚❚
        </span>
      </button>
      <button
        class="sc-fznKkj kWHrLL"
        type="submit"
        width="100"
      >
        <span
          class="sc-fzplWN gcyZgM"
        >
          ▷
        </span>
      </button>
      <button
        class="sc-fznKkj kWHrLL"
        type="submit"
        width="100"
      >
        <span
          class="sc-fzplWN gcyZgM"
        >
          ↻
        </span>
      </button>
    </div>
    <div
      class="sc-AxjAm dNRMDR"
    >
      <div
        class="sc-AxirZ khsKjH"
      >
        <input
          class="sc-fzpans dJYYuJ"
          id="todoName"
          placeholder="Add todo"
          value=""
        />
        <button
          class="sc-fznKkj bHKJNM"
          type="submit"
        >
          <span
            class="sc-fzplWN gcyZgM"
          >
            Add Todo
          </span>
        </button>
      </div>
      <div
        class="sc-AxirZ dfLLgI"
      >
        <div
          style="padding: 47px 15px 0px;"
        >
          <button
            class="sc-fznKkj eboykN"
            type="submit"
            width="100"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              Auto
            </span>
          </button>
          <button
            class="sc-fznKkj kCeHpr"
            type="submit"
            width="120"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              Pomodoro
            </span>
          </button>
          <button
            class="sc-fznKkj kCeHpr"
            type="submit"
            width="120"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              L-Break
            </span>
          </button>
          <button
            class="sc-fznKkj kCeHpr"
            type="submit"
            width="120"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              S-break
            </span>
          </button>
          <button
            class="sc-fznKkj eboykN"
            type="submit"
            width="100"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              Settings
            </span>
          </button>
          <button
            class="sc-fznKkj eboykN"
            type="submit"
            width="100"
          >
            <span
              class="sc-fzplWN gcyZgM"
            >
              About
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-AxiKw hWCQXE"
    >
      <button
        class="sc-fznyAO jbexnx"
        type="submit"
        width="150"
      >
        Show Favorites
      </button>
      
      <table>
        <thead>
          <tr>
            <th />
            <th>
              Name
            </th>
            <th>
              Duration
            </th>
            <th>
              L-break
            </th>
            <th>
              S-break
            </th>
            <th>
              Status
            </th>
            <th>
              Favorites
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class=""
            data-position="0"
            draggable="true"
          >
            <td>
              <p
                class="sc-Axmtr cuKjKQ"
                type="p"
              >
                ✎
              </p>
            </td>
            <td>
               
              Taena hehe
            </td>
            <td>
               
              25
            </td>
            <td>
               
              15
            </td>
            <td>
               
              10
            </td>
            <td>
              To do
            </td>
            <td>
               
              No
            </td>
            <td>
              <button
                class="sc-fznyAO jqcObA"
                type="submit"
                width="100"
              >
                Play
              </button>
              <button
                class="sc-fznyAO bLTcgU"
                type="submit"
                width="100"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr
            class="done"
            data-position="1"
            draggable="true"
          >
            <td>
              <p
                class="sc-Axmtr cuKjKQ"
                type="p"
              >
                ✎
              </p>
            </td>
            <td>
               
              Taena KO
            </td>
            <td>
               
              25
            </td>
            <td>
               
              15
            </td>
            <td>
               
              10
            </td>
            <td>
              Done
            </td>
            <td>
               
              Yes
            </td>
            <td>
              <button
                class="sc-fznyAO jqcObA"
                disabled=""
                type="submit"
                width="100"
              >
                Play
              </button>
              <button
                class="sc-fznyAO bLTcgU"
                type="submit"
                width="100"
              >
                Delete
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </form>
</div>
`;
